src/avram/where.cr

```diff
@@ -1,31 +1,52 @@
 module Avram::Where
-  abstract class SqlClause
-    getter :column
-    getter :value
+  abstract class Condition
+    abstract def prepare(placeholder_supplier : Proc(String)) : String
 
-    def initialize(@column : Symbol | String, @value : String | Array(String) | Array(Int32))
+    def ==(other : Condition)
+      prepare(->{ "unused" }) == other.prepare(->{ "unused" })
     end
 
-    abstract def operator : String
-    abstract def negated : SqlClause
-
-    def prepare(prepared_statement_placeholder : String)
-      "#{column} #{operator} #{prepared_statement_placeholder}"
+    def ==(other)
+      false
     end
 
     def clone
       self
     end
   end
 
-  abstract class NullSqlClause < SqlClause
-    @value = "NULL"
+  abstract class SqlClause < Condition
+    getter column : Symbol | String
 
-    def initialize(@column : Symbol | String)
+    def initialize(@column)
     end
 
-    def prepare
-      "#{column} #{operator} #{@value}"
+    abstract def operator : String
+    abstract def negated : SqlClause
+
+    def prepare(placeholder_supplier : Proc(String)) : String
+      "#{column} #{operator} #{placeholder_supplier.call}"
+    end
+  end
+
+  abstract class ValueHoldingSqlClause < SqlClause
+    getter value : String | Array(String) | Array(Int32)
```

Comment:
I know this type comes from what was defined in Condition before, but any idea why these types specifically? I ask because I'm wondering if with the latest pg and db, will this blow up when someone uses UUID? 🤔 Also, Int64 I guess... I'm assuming this is the value of a DB column which could be many different types... Or maybe I'm wrong and this value is something else?

Reply:
I'm not able to hunt down why we convert everything to these types

ultimately, we call out to here https://github.com/crystal-lang/crystal-db/blob/bd456028644853f852f5adfe053142e6ad885e52/src/db/statement.cr#L69-L71
which does not have type signatures so it's not very helpful.  Our process ultimately delegates to the `#to_db` method for each type which in most things converts to a string (Ints, UUIDs, etc).

Reply:
It'll probably be ok, but lets keep this in mind if we notice something weird popup. I feel like this should have more types on it, but if it works as is then we can just roll with it.